### PR TITLE
models.py updated

### PR DESCRIPTION
**Pull request by shirishamuthineni**

### CODE REVIEW

File: FusionIIIT/applications/scholarships/models.py [hunk 1]
```diff
@@ -1,3 +1,150 @@
...
+                         ('business','Business'),
+                         ('medical','Medical'),
+                         ('consultant','Consultant'),
+                        )
+            ),
+            ('pensioners','Pensioners'),
+    )
+    father_occ_choice=models.CharField(max_length=10,choices=FATHER_OCC_CHOICES)
+    father_occ          = models.TextField(max_length=100)
+    EMPLOYED='EMP'
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: no need to give variable names. You can directly use this in the dictionary of choices. Refer to models in global for reference.

File: FusionIIIT/applications/scholarships/models.py [hunk 2]
```diff
@@ -1,3 +1,150 @@
...
+    house_type         = models.CharField(max_length=5,choices=HOUSE_TYPE_CHOICES)
+    house_area         = models.IntegerField()
+    sSchool_10          = models.CharField(max_length=40)
+    school_10_fee      = models.IntegerField()
+    school_12          = models.CharField(max_length=40)
+    school_12_fee      = models.IntegerField()
+    father_declation   = models.FileField(null=True, blank=True)
+    affidavit          = models.FileField(null=True, blank=True)
+    
+    COMPLETE='CMP'
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Same for all the keys of dictionary.

File: FusionIIIT/applications/scholarships/models.py [hunk 3]
```diff
@@ -1,3 +1,150 @@
...
+    two_wheeler        = models.IntegerField()
+    RENTED='rent'
+    OWNES='own'
+    HOUSE_TYPE_CHOICES =(
+                        (RENTED,'rented'),
+                        (OWNED,'owned'),
+    )
+    house_type         = models.CharField(max_length=5,choices=HOUSE_TYPE_CHOICES)
+    house_area         = models.IntegerField()
+    sSchool_10          = models.CharField(max_length=40)
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Spelling mistake

File: FusionIIIT/applications/scholarships/models.py [hunk 4]
```diff
@@ -1,3 +1,150 @@
 # from django.db import models
 
 # Create your models here.
+from django.db import models
+from applications.academic_information.models import Student
+# Create your models here
+
+class Mcm(models.Model):
+    mcm_id              = models.IntegerField(primary_key=True)
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Remove this as default will serve better.

File: FusionIIIT/applications/scholarships/models.py [hunk 5]
```diff
@@ -1,3 +1,150 @@
...
+    )
+    batch = (
+        ('UG1','UG1'),
+        ('UG2','UG2'),
+        ('UG3','UG3'),
+        ('UG4','UG4'),
+        ('PG1','PG1'),
+        ('PG2','PG2')
+    )
+
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: This is correct. You can make key value pairs without the need of declaring variables for each.

File: FusionIIIT/applications/scholarships/models.py [hunk 6]
```diff
@@ -1,3 +1,150 @@
...
+
+
+
+class Award_and_scholarship(models.Model):
+    award_id = models.IntegerField(primary_key=True)
+    award_name = models.CharField(max_length=50, default='') 
+    catalog = models.TextField()
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: max length as 1000

File: FusionIIIT/applications/scholarships/models.py [hunk 7]
```diff
@@ -1,3 +1,150 @@
...
+
+    class Meta:
+        db_table = 'Award_and_scholarship'
+        
+    def __str__(self):
+        return str(self.award_id)
+
+class Previous_winner(models.Model):
+    previous_id = models.IntegerField(primary_key=True)
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Default can be used here as well

File: FusionIIIT/applications/scholarships/models.py [hunk 8]
```diff
@@ -1,3 +1,150 @@
...
+    student_id = models.ForeignKey(Student)
+    award_id = models.ForeignKey(Award_and_scholarship)
+    class Meta:
+        db_table = 'Previous_winner'
+        
+    def __str__(self):
+        return self.previous_id
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Add year field for the year the award/scholarship was received.

File: FusionIIIT/applications/scholarships/models.py [hunk 9]
```diff
@@ -1,3 +1,150 @@
...
+
+class Release(models.Model):
+    release_id = models.IntegerField(primary_key=True)
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: No need for this either. Use default.

**Review by arshadsid — CHANGES_REQUESTED** [review]
Take a look at comments I added . Also, make it flake8 and isort compliant.

File: FusionIIIT/applications/scholarships/models.py (hunk 6 above)
Comment (shirishamuthineni) [reply FusionIIIT/applications/scholarships/models.py]: 100 or 1000

Comment (arshadsid) [reply FusionIIIT/applications/scholarships/models.py]: 1000

File: FusionIIIT/applications/scholarships/models.py (hunk 2 above)
Comment (shirishamuthineni) [reply FusionIIIT/applications/scholarships/models.py]: what is wrong in this

File: FusionIIIT/applications/scholarships/models.py [hunk 10]
```diff
@@ -1,3 +1,170 @@
...
+    batch = (
+        ('UG1','UG1'),
+        ('UG2','UG2'),
+        ('UG3','UG3'),
+        ('UG4','UG4'),
+        ('PG1','PG1'),
+        ('PG2','PG2')
+    )
+class Mcm(models.Model):
+    mcm_id              = models.IntegerField()
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Use default id. Delete this field

File: FusionIIIT/applications/scholarships/models.py [hunk 11]
```diff
@@ -1,3 +1,170 @@
...
+        
+    def __str__(self):
+        return self.mcm_id
+
+
+
+
+
+class Award_and_scholarship(models.Model):
+    award_id           = models.IntegerField()
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Use default id. Delete this field

File: FusionIIIT/applications/scholarships/models.py [hunk 12]
```diff
@@ -1,3 +1,170 @@
...
+    catalog            = models.TextField()
+
+    class Meta:
+        db_table       = 'Award_and_scholarship'
+        
+    def __str__(self):
+        return str(self.award_id)
+
+class Previous_winner(models.Model):
+    previous_id        = models.IntegerField(primary_key=True)
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Use default id. Delete this field

File: FusionIIIT/applications/scholarships/models.py [hunk 13]
```diff
@@ -1,3 +1,170 @@
...
+    award_id           = models.ForeignKey(Award_and_scholarship)
+    year               = models.DateTimeField(default=datetime.now().year)
+    class Meta:
+        db_table       = 'Previous_winner'
+        
+    def __str__(self):
+        return self.previous_id
+
+class Release(models.Model):
+    release_id        = models.IntegerField()
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Use default id. Delete this field

**Review by arshadsid — CHANGES_REQUESTED** [review]
Make the above mentioned changes.
Also, the commit is not flake8 and isort compliant.

**Comment by arshadsid** [comment]
@shirishamuthineni have a look at the comments. Also, make ot isort and flake8 compliant.

**Comment by Samy-33** [comment]
Check has failed, please refer to last line of `README.md`.  
Run those two scripts before pushing.

**Comment by Samy-33** [comment]
@shirishamuthineni , are you running `flake8`  and `isort` commands given in readme, before pushing?  
If not then run them and resolve errors accordingly.

File: FusionIIIT/applications/scholarships/models.py (hunk 8 above)
Comment (shirishamuthineni) [reply FusionIIIT/applications/scholarships/models.py]: ha added

**Review by arshadsid — CHANGES_REQUESTED** [review]
Use default model ids.

File: FusionIIIT/applications/scholarships/models.py [hunk 14]
```diff
@@ -1,3 +1,269 @@
...
+        ('UG2', 'UG2'),
+        ('UG3', 'UG3'),
+        ('UG4', 'UG4'),
+        ('PG1', 'PG1'),
+        ('PG2', 'PG2')
+    )
+
+
+class Mcm(models.Model):
+    email_id = models.CharField(max_length=25)
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py]: Student id has to remain!

File: FusionIIIT/applications/scholarships/models.py [hunk 15]
```diff
@@ -1,3 +1,269 @@
...
+
+class Award_and_scholarship(models.Model):
+    award_name = models.CharField(max_length=1000, default='')
+    catalog = models.TextField()
+
+    class Meta:
+        db_table = 'Award_and_scholarship'
+
+
+class Previous_winner(models.Model):
```
Comment (arshadsid) [inline FusionIIIT/applications/scholarships/models.py:129]: Where did the student go!

**Review by arshadsid — CHANGES_REQUESTED** [review]
You had to remove the primary keys which were auto-generated as the default solves the purpose.. You removed every field that had id in it!
Where did the student go?

**Review by arshadsid — CHANGES_REQUESTED** [review]
Close your pull request. Consult your team lead and update according to the schema.

File: FusionIIIT/applications/scholarships/models.py (hunk 14 above)
Comment (shirishamuthineni) [reply FusionIIIT/applications/scholarships/models.py]: We are traveling right now in train neccesary changes will be done tomorrow

**Pull request closed without merging**